### PR TITLE
Updating Template to call set_value in the setter function

### PR DESCRIPTION
**Pull request by sourabhdalvi**

### CODE REVIEW

**Review by jd-lara — APPROVED** [review]
we can merge and tag when @daniel-thom approves

**Review by daniel-thom — APPROVED** [review]
LGTM

File: src/utils/generate_structs.jl [hunk 1]
```diff
@@ -66,7 +66,7 @@ end
 
 {{#setters}}
 {{#create_docstring}}\"\"\"Set [`{{struct_name}}`](@ref) `{{name}}`.\"\"\"{{/create_docstring}}
-{{setter}}(value::{{struct_name}}, val) = value.{{name}} = val
+{{setter}}(value::{{struct_name}}, val) = value.{{name}} = {{#needs_conversion}}set_value(value, val){{/needs_conversion}}{{^needs_conversion}}val{{/needs_conversion}}
```
Comment (daniel-thom) [inline src/utils/generate_structs.jl:69]: `set_value!`

**Comment by daniel-thom** [comment]
Can you update the docs for this option?
```
docs/src/dev_guide/auto_generation.md
```

**Comment by sourabhdalvi** [comment]
> Can you update the docs for this option?
> 
> ```
> docs/src/dev_guide/auto_generation.md
> ```

Updated the `needs_conversion` section

**Review by daniel-thom — APPROVED** [review]

**Pull request merged**